secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  RecoveryPhraseStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/10/2022.
+//
+
+import Foundation
+
+/// Zcash implementation of the keychain that is not universal but designed to deliver functionality needed by the wallet itself.
+/// All the APIs should be thread safe according to official doc:
+/// https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
+// swiftlint:disable convenience_type
+final class RecoveryPhraseStorage {
+    enum Constants {
+        static let zcashStoredWallet = "zcashStoredWallet"
+        /// Versioning of the stored data
+        static let zcashKeychainVersion = 1
+    }
+
+    enum KeychainError: Error, Equatable {
+        case decoding
+        case duplicate
+        case encoding
+        case noDataFound
+        case unknown(OSStatus)
+        case unsupportedVersion(Int)
+    }
+
+    static let `default`: RecoveryPhraseStorage = RecoveryPhraseStorage()
```

Comment:
Do we need this to be a singleton?  Is there a reason we can't rely on dependency injection?

Reply:
thinking about it, you are right, there's no special value in the singleton, removed